### PR TITLE
Finish up the conversion to Kotlin

### PR DESCRIPTION
**Pull request by nkrijnen**
@emartynov very nice that you took the initiative to convert it to Kotlin, I was playing with the same idea... now that you made a start, I couldn't resist to finish it up.

Some more things I tweaked:
* Rename Java to Kotlin in some places
* Simplify the INSTALLATION instruction to get the tests running
* Use kotlin.test imports instead of JUnit (had to convert to gradle to get that to work - but as most kotlin projects use gradle that probably makes sense anyway)

Curious what you think of it.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,15 +2,10 @@ target/
...
 
 ### IntelliJ IDEA ###
-.idea/modules.xml
-.idea/jarRepositories.xml
-.idea/compiler.xml
-.idea/libraries/
-*.iws
-*.iml
-*.ipr
+.idea
```
Comment (emartynov) [inline .gitignore:8]: Thanks for updating it. I was lazy to finally decide what files I wanted to have in repo from idea folder and what I did not.

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+plugins {
+    kotlin("jvm") version "1.6.20"
```
Comment (emartynov) [inline build.gradle.kts:2]: Please update to the latest stable (1.8.10)?

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    api("org.joda:joda-money:1.0.1")
+    implementation(kotlin("stdlib"))
```
Comment (emartynov) [inline build.gradle.kts:11]: With new kotlin plugin this is not needed.

File: pom.xml
Comment (emartynov) [inline pom.xml:1]: Please keep it - so people that what to use maven still can use the project. There are also github actions that are using maven to run tests on CI.

File: src/test/kotlin/com/carriesCars/pricingEngine/DurationTest.kt [hunk 4]
```diff
@@ -1,23 +1,22 @@
...
+import kotlin.test.assertFailsWith
 
 class DurationTest {
     @Test
     fun `Duration is at least one minute`() {
-        assertThrows(IllegalArgumentException::class.java) {
+        assertFailsWith<IllegalArgumentException> {
             val lessThan1 = 0L
-            val duration = ofMinutes(lessThan1)
+            Duration.ofMinutes(lessThan1)
```
Comment (emartynov) [inline src/test/kotlin/com/carriesCars/pricingEngine/DurationTest.kt:13]: Thanks for this - I should have the scope of the function clear and not to have import for it.

**Review by emartynov — COMMENTED** [review]
Thank you so much for the cleanup and polishing of all project bits.

I don't feel the necessity to use kotlin tests since junit works fine until you are not KMP.

I also didn't feel the necessity to change the build system, and it is nice to have them both. Please add a gradle wrapper to the project so people unfamiliar with gradle don't need to learn how to install gradle on the machine (this is also the preferred way). Please keep maven in the project so people can decide which type of project to import into Idea.

Otherwise, it looks great to me. I dropped a couple of thank you comments. Super thank you for the contribution.

**Comment by emartynov** [comment]
After merging this PR I will rebase the `problems` branch and fix all conflicts.

File: build.gradle.kts (hunk 2 above)
Comment (nkrijnen) [reply build.gradle.kts:2]: Good one, forgot to do that after I copied it :)

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
```
Comment (emartynov) [inline gradle/wrapper/gradle-wrapper.properties:3]: Can you also use the latest here? (8.0.2)

**Comment by emartynov** [comment]
@nkrijnen I will merge it and do quick changes myself. Super thank you for the contribution!

**Pull request merged**